astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -257,6 +257,10 @@ def _arithmetic(self, operation, operand,
                 operation, operand, result, uncertainty_correlation,
                 **kwds2['uncertainty'])
 
+        # If both are None, there is nothing to do.
+        if self.psf is not None or operand.psf is not None:
+            warnings.warn(f"Not setting psf attribute during {operation.__name__}.", RuntimeWarning)
```

Comment:
You'll have to add the import yourself. I cannot suggest it at the line where it is supposed to go. Now precommit complains if it is not at the right place.

```suggestion
            warnings.warn(f"Not setting psf attribute during {operation.__name__}.", AstropyUserWarning)
```